### PR TITLE
Feature/fix when one module fail all fails

### PR DESCRIPTION
**Pull request by thallesfreitaszup**
## Issue
before when one component of module fail all modules of the deployment  was being set to status failed 

## Solution
set status failed only the module of the component that failed

### CODE REVIEW

**Comment by thallesfreitaszup** [comment]
[Na falha de um modulo, o status de todos os outros modulos do deployment é setado para FAILED](https://trello.com/c/s9BENEkd/494-na-falha-de-um-modulo-o-status-de-todos-os-outros-modulos-do-deployment-%C3%A9-setado-para-failed)

File: butler/src/app/core/services/deployments/status-management-service.ts [hunk 1]
```diff
@@ -418,4 +411,10 @@ export class StatusManagementService {
         await this.propagateFailedStatusChangeToModule(componentDeploymentEntity.moduleDeployment)
         await this.propagageFailedStatusChangeToDeployment(componentDeploymentEntity.moduleDeployment.deployment)
     }
+
+    public async deepUpdateComponentStatus(componentDeployment: ComponentDeploymentEntity): Promise<void> {
```
Comment (lucasbfernandes) [inline butler/src/app/core/services/deployments/status-management-service.ts:415]: Maybe this method could have a more meaningful name? It is not doing any deep updates.

File: butler/src/app/core/services/deployments/status-management-service.ts [hunk 2]
```diff
@@ -81,15 +81,8 @@ export class StatusManagementService {
...
-    public async deepUpdateDeploymentStatus(deployment: DeploymentEntity, status: DeploymentStatusEnum) : Promise<void[][]>{
-        await this.deploymentsRepository.updateStatus(deployment.id, status)
-        if (!deployment.modules) {
-            deployment.modules =
-                await this.moduleDeploymentRepository.find({
-                    where: { deployment: {id: deployment.id} }
-                })
-        }
-        return Promise.all(deployment.modules.map(m => this.deepUpdateModuleStatus(m, status)))
+    public async deepUpdateDeploymentStatus(deployment: DeploymentEntity, status: DeploymentStatusEnum) : Promise<void>{
```
Comment (lucasbfernandes) [inline butler/src/app/core/services/deployments/status-management-service.ts:84]: Maybe this method could have a more meaningful name? It is not doing any deep updates.

File: butler/src/app/core/services/deployments/status-management-service.ts [hunk 3]
```diff
@@ -81,15 +81,8 @@ export class StatusManagementService {
         )
     }
 
-    public async deepUpdateDeploymentStatus(deployment: DeploymentEntity, status: DeploymentStatusEnum) : Promise<void[][]>{
```
Comment (lucasbfernandes) [inline butler/src/app/core/services/deployments/status-management-service.ts:84]: You should also delete methods that will become unnecessary after this deletion

**Pull request closed without merging**